### DISCUSSION TITLE
Create entitlement driven views

### DISCUSSION THREAD

**Discussion by ThatNinjaGuy** [ThatNinjaGuy/Waiter#109] [Ideas]

**Comment by ThatNinjaGuy** [comment]
Create list of features based on individual roles and screens to be hidden/ visible

**Comment by imsudiproy** [comment]
Hi @ThatNinjaGuy , can you please provide more datails?

**Comment by ThatNinjaGuy** [comment]
On the various screens, the current functionality is to show everything for a particular set of users. But it straight away opens an unauthorized screen component for the users who don't have access. This restricts the non-admin users from even seeing the menu/staffs/inventory screen. This should not be desirable. Users should be able to see everything. Although the functionality on those screens should be restricted. Like, a non-admin user should not be allowed to edit details for the staffs. This can create issues of misuse by the users. Similarly, features need to be identified which can be enabled/disabled on the screens.

**Comment by ThatNinjaGuy** [comment]
In the current scope which should be developed, the entitlement driven views have been decided to be as follows:
The functionality being implemented is to restrict any update/delete operation by non-admin users on Staffs/Inventory/Menu screens.
On orders page, entitlement levels till kitchen staffs should be able to accept orders in kitchen, entitlement levels till waiter should be able to pick up. And similarly entitlement level till Waiter should be able to take order. Signup requests can only be updated by admin users. New features to enable can be suggested.

**Comment by ThatNinjaGuy** [comment]
Fixed basic functionality.
Pending task: On orders page, entitlement levels till kitchen staffs should be able to accept orders in kitchen, entitlement levels till waiter should be able to pick up.
More enhancements could be done, but features need to be identified.
